modules/rgbd/src/normal.cpp

```diff
@@ -129,13 +129,22 @@ namespace rgbd
   signNormal(const Vec<T, 3> & normal_in, Vec<T, 3> & normal_out)
   {
     Vec<T, 3> res;
-    if (normal_in[2] > 0)
-      res = -normal_in / norm_vec(normal_in);
+    if(norm_vec(normal_in) == 0) { //avoid division by 0
+      normal_out[0] = 0;
```

Comment:
I don't think I follow what are you saying... When you talk about "unknown depth" do you mean that if the input is a 16U depth than the normal output should have 0 in the undefined locations or NaN when the input is 32F or 64F? I suppose this is the aim of `quiet_NaN`.

Anyway it means that the user needs to handle NaN values in the output Mat, leading to unusual (maybe just for me) checks, for example when quantizing the normals in the linemod case (by the way I intended to send a PR to replace the actual `quantizedNormals` function to use this normal computer)